lambdaclass/mirra_backend
# [GH-727] Fix match register for old users

## Motivation

Every time you played a match they wasn't being saved in the backend. This was due to the `create_arena_match_results` failing on the `give_prestige` step due to old users not having the unit to update. We have the following error in the backend:

``` 
** (exit) an exception was raised:
Jun 27 19:21:13 mirra-central-europe-testing entrypoint.sh[1388105]:     ** (CaseClauseError) no case clause matching: {:error, {:update_prestige, "a8875c87-7e09-48e3-9ab5-06e059a6a69e"}, :unit_not_found,
```

The behavior should be to create the unit if that user doesn't have it, this would be an issue in the future when we add new units to the db

Closes #727

## Summary of changes

- Create unit instead on failing when not present in the user when running `create_arena_match_results`

## How to test it?

- Log in as Guest
- Run this command `GameBackend.Repo.delete_all(GameBackend.Units.Unit)`
- Play a match and it should be registered

## Checklist
